hack/test-go.sh

```diff
@@ -77,22 +77,45 @@ fi
 
 KUBE_TIMEOUT=${KUBE_TIMEOUT:--timeout 60s}
 
-if [ "${1-}" != "" ]; then
-  if [[ "${1}" == *"/..." ]]; then
+# Declare and set arrays to null string. We need to always keep the first
+# element to be the null string, otherwise the variable is considered unset.
+# When using, slice it to ignore the first element: ${varname[@]:1}.
+declare -a test_flags= test_packages=
```

Comment:
I think we should rethink how we pass arguments to `test-go`. Currently we're doing too much and it's pretty ridiculous. I suggest we pass *everything* that isn't a package to test through bash variables `GO_TEST_FLAGS='whatever' hack/test-go.sh <packages>` and only pass packages as args.